tests/rptest/util.py

```diff
@@ -53,7 +53,7 @@ def release(self):
         return self._scale == Scale.RELEASE
 
 
-def wait_until_result(condition, *args, **kwargs):
+def wait_until_result(condition, *args, **kwargs) -> Unknown:
```

Comment:
Neat trick. Does this mean that the type checker won't scream about `None` checks on the value returned from `wait_until_result`?

Reply:
Yupp. Although it'd be nice to codify the actual method, but I don't have enough python typing foo in me (maybe I'm nerdsniped now and I'll send a followup PR)

Reply:
Looks like we would need python 3.11: https://peps.python.org/pep-0646/

Reply:
So sick. I hate that None check. Death by a thousand meaningless None checks 🙃

Reply:
I wish there was a variant of python that required type checks to pass. Maybe we would have to setup pyright formally or something...

Reply:
>maybe we would have to setup pyright formally

Yeah. No clue whether that can be done piecemeal, but the equivalence of "no type hints present" to "type checks pass" makes the whole thing less useful IMO. Too easy to forget.

Reply:
Ugh needs to be Any not Unknown 🤦

Reply:
Love it. 🐍 🕳️